tests/sandbox/LegacyForwardIterator.h

```diff
@@ -0,0 +1,99 @@
+// FIXME remove pragma after moving to cts
+
+#pragma once
+
+#include "../common/common.h"
+
+#include "LegacyInputIterator.h"
+#include "LegacyOutputIterator.h"
+#include "TypeTraits.h"
+#include "common.h"
+
+#include <type_traits>
+#include <utility>
+
+template <typename It>
+class legacy_forward_iterator_requirement : requirement_verifier {
+ public:
+  bool check(It valid_iterator, size_t vector_size,
+             const std::string type_name) {
+    INFO("Verify named requiremnt Legacy Input Iterator for: " + type_name);
+    STATIC_CHECK(!std::is_same_v<It, void>);
+
+    legacy_input_iterator_requirement<It>{}.check(type_name);
+
+    {
+      INFO("Iterator have to be default constructible");
+      verify(std::is_default_constructible_v<It>);
+    }
+
+    using iterator_reference = typename std::iterator_traits<It>::reference;

```

Comment:
Will fail a compilation of reference_t is not defined. There is similar solution in `LegacyInputIterator`, that will be safety.